src/Trip.js

```diff
@@ -1,17 +1,33 @@
 class Trip {
   constructor(trip, destinations) {
-    this.id = trip["id"] || Date.now()
-    this.userID = trip["userID"]
-    this.destination = destinations.find(destination => destination.id === trip["destinationID"]) || trip["destinationID"]
-    this.travelers = trip["travelers"]
-    this.date = trip["date"]
-    this.duration = trip["duration"]
-    this.status = trip["status"] || "pending"
-    this.suggestedActivities = trip["suggestedActivities"]
+    this.id = trip.id || Date.now()
+    this.userID = trip.userID
+    this.getDestination(destinations, trip.destinationID)
+    this.travelers = trip.travelers
+    this.date = trip.date
+    this.duration = trip.duration
+    this.status = trip.status || "pending"
+    this.suggestedActivities = trip.suggestedActivities
+    this.setEndDate() 
```

Comment:
Have you come across any problems yet with calling a Class method inside of its own constructor? It seems savvy, but I'd also be concerned about errors occurring due to the fact that when initialized the method as yet to be defined (I'm thinking about the single-threadedness of Javascript with this).